### ISSUE TITLE
V3 Radio Group cannot be accessed via Keyboard when NVDA or JAWS are in use

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/vets-design-system-documentation#2035]
# Bug Report

- [X] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

While using JAWS and NVDA on a PC to test the [Alternate Signer form](https://staging.va.gov/supporting-forms-for-claims/alternate-signer-form-21-0972/introduction) (which uses the v3 radio group component), I was not able to use the keyboard to select different radio options in a predictable and reliable way. 

I then tried the [component in storybook](https://design.va.gov/storybook/iframe.html?id=uswds-va-radio--default&viewMode=story), and the same error happened in storybook.

I also tried the [USWDS Radio buttons](https://designsystem.digital.gov/components/radio-buttons/) and the issue did not happen there.

Here is a video capturing the issue:

https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/117098918/1fd666d3-d8c0-420c-8e85-d305d0bb6968

At the beginning of the video you'll hear the radio labels being read aloud, but no visual indicator that anything is changing. I can hit the spacebar and it will select an option, but if I want to change that option by using the keyboard and going up and down, again, the audio changes, but the visuals do not.

In the video you then see me delete the attribute `usdws="true"` from the radio buttons, and I can then use the keyboard to go up and down through the radio buttons, and see which one is being selected. 

Finally, when I turn off the screen reader, and refresh the page, (so that `usdws="true"` is back, I can then go up and down through the radio buttons using a keyboard.

While using JAWS, a similar behavior happens, but when I hit the space bar to select a radio button, nothing has been indicated to the user that a selection was made, and it is only through tabbing back through and landing on that selected one again that you know it has been selected through audio.

## What I expected to happen

I expect to be able to navigate through the radio group using the keyboard. Tabbing into the group then using the keyboard up and down arrows to select the radio button I want. I expect to see a visual indication to represent the radio button I'm on and the screen reader to read the label at the same time.

## Reproducing
- Formation version:
- Device: (e.g. iPhone 8, Macbook) Windows
- Browser: (e.g. Firefox, IE 11) Chrome
- Screen reader: NVDA

Steps to reproduce:

1. From windows, start NVDA or JAWS
2. Navigate in a browser to https://design.va.gov/storybook/iframe.html?id=uswds-va-radio--default&viewMode=story
3. Tab into the viewport, hear the question  "Select one historical figure read aloud" and try to use the arrow keys to select different radio options
4. While the audio changes, there's no visual indicator to match what is being read aloud. No indication of what is being selected.

## Urgency
How urgent is this request? Please select the appropriate option below and/or provide details
- [X] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

### Details
The v1 component works as expected when using an NVDA screen reader, so this feels like a regression backwards in its usability.

**Comment by caw310** [comment]
This was planned for sprint 35 not 34 although Ryan got a start on this at the end of sprint 34.

**Comment by rsmithadhoc** [comment]
Quick update on this:

- We have custom code to switch to the next item using arrow keys. Without NVDA on, this code runs without any issue. When NVDA is on, we never get to this code when using arrow keys. Normally, when NVDA sees a standard HTML radio group, arrow keys will go to the next item. I don't believe NVDA is seeing it as a radio group, so it is intercepting the key press and performing a different action. In this case, it is saying the next character.
- While I'm not sure the exact fix yet, we may have to modify the structure of the component to ensure that screen readers are correctly recognizing this as a radio group. This component is a bit more complex than its standard HTML counterpart, so it may take some trial and error.

**Comment by rsmithadhoc** [comment]
- What is being cycled through via the arrow keys are the `va-radio-option` elements. Because those are custom elements, the screen reader doesn't know those are inputs, so it performs its default action.
- To fix this, I believe we need to rework the following to use the `<input>`, rather than the custom element. I created a [pull request](https://github.com/department-of-veterans-affairs/component-library/pull/838) with some initial work that gets screen readers working, but there is more to do to ensure the component works as intended. I also attached a video of this with the screen reader.
- This may change some core things about this component, so we will want to make sure other things still function as intended. This includes, but is not limited to:
    - Focus and its indicator is properly set.
    - Tabbing to the radio group starts on the first item.
    - The correct radio item gets selected.
    - Moving forward and backward through the list works as intended.
    - When moving to the next radio button, the option label is read. "Radio button 1 of 4, Sojourner Truth" and not just "Radio button 1 of 4".
    
I created a new issue https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2075 for this implementation and closing out this for the research portion.
